### PR TITLE
Fix SELinux permissive mode detection logic

### LINKED ISSUE #135: SELinux permissive detection causes re-provisioning of nodes
Currently `wksctl` requires each node to be in the SELinux permissive mode.
So there's a detection of this mode before building a plan.

There's a scenario that a node did not install any SELinux package in the first place.
The the detection checked that there's no any SELinux command, so it decided to not run `selinux:permissive` step.

After that, the first plan was built, and added to the node.
Later, Docker was installed on that node which also installed SELinux as a Docker dependency. Now there existed a SELinux command.
The detection checked the node again and said, "hey this node needs an update. We needs to run `selinux:permissive` step on every node".
And the controller started to provision every node again. This result was in collapsing the seed node and rendering the whole cluster to the failure state and the control plane's also gone.

### PR DESCRIPTION
**Pull request by chanwit**
This PR improves the detection of SELinux permissive mode to prevent node re-provision.
It requires #128 to be merged.

Fixes #135

### CODE REVIEW

File: pkg/plan/recipe/install_plans.go [hunk 1]
```diff
@@ -108,6 +111,17 @@ func BuildCRIPlan(criSpec *baremetalspecv1.ContainerRuntime, cfg *envcfg.EnvSpec
 			plan.DependOn("install:docker"))
 	}
 
+	if IsDockerOnCentOS {
```
Comment (palemtnrider) [inline pkg/plan/recipe/install_plans.go]: This duplicates what is in BuildK8SPlan.  Should that depend on install:docker or should that other block be deleted?

Comment (palemtnrider) [reply pkg/plan/recipe/install_plans.go]: Thoughts?

Comment (jrryjcksn) [reply pkg/plan/recipe/install_plans.go]: This isn't exactly a duplicate. The original case _conditionally_ adds the resource based on a probe of the machine environment:

File: pkg/apis/wksprovider/controller/wksctl/machine_actuator.go [hunk 2]
```diff
@@ -433,7 +433,7 @@ func (a *MachineActuator) update(ctx context.Context, cluster *clusterv1.Cluster
 
 	if diffedPlan, err := plandiff.GetUnifiedDiff(currentPlan, planJSON); err == nil {
 		contextLog.Info("........................ DIFF PLAN ........................")
-		contextLog.Info(diffedPlan)
+		fmt.Print(diffedPlan)
```
[the remark this answers is not in the document]
Comment (jrryjcksn) [reply pkg/apis/wksprovider/controller/wksctl/machine_actuator.go]: I confirmed that the diff is illegible when logged...

File: pkg/plan/recipe/install_plans.go (hunk 1 above)
Comment (chanwit) [reply pkg/plan/recipe/install_plans.go]: I'm addressing this change. thank you @jrryjcksn !

Comment (chanwit) [reply pkg/plan/recipe/install_plans.go]: I'm fixing the logic. But these 2 selimux:permissive steps are *slightly* different.

**Comment by chanwit** [comment]
@palemtnrider @jrryjcksn I'm going to re-test this patch with both Footloose and Ignite setups.
I'll ping you again after I get results.

File: pkg/plan/recipe/install_plans.go [hunk 3]
```diff
@@ -108,6 +111,20 @@ func BuildCRIPlan(criSpec *baremetalspecv1.ContainerRuntime, cfg *envcfg.EnvSpec
 			plan.DependOn("install:docker"))
 	}
 
+	// this is a special case: if SELinux is not there on RH, CentOS Linux family
+	// installing Docker will also installing SELinux
+	// then we set SELinux mode to be permissive right after the docker installation step
+	if IsDockerOnCentOS && cfg.SetSELinuxPermissive {
```
Comment (chanwit) [inline pkg/plan/recipe/install_plans.go]: ping @palemtnrider @jrryjcksn

File: pkg/utilities/envcfg/envcfg_test.go [hunk 4]
```diff
@@ -105,12 +111,14 @@ func TestGetEnvSpecificConfig(t *testing.T) {
...
-					cmdSELinuxEnabled: {},
-					cmdMachineID:      {out: "01234567"},
-					cmdUUID:           {out: "01234567"},
+					cmdOsRel:             {out: relCentos},
+					cmdEnv:               {out: envContainer},
+					cmdSELinuxFound:      {},
+					cmdSELinuxEnabled:    {},
+					cmdMachineID:         {out: "01234567"},
+					cmdUUID:              {out: "01234567"},
+					cmdSELinuxPermissive: {}, //, err: &plan.RunError{ExitCode: 1}},
```
Comment (jrryjcksn) [inline pkg/utilities/envcfg/envcfg_test.go]: Nit... leftover comment?

**Comment by chanwit** [comment]
re-triggered the CI

**Comment by chanwit** [comment]
@palemtnrider @jrryjcksn ptal again
I cannot merge it myself

**Comment by jrryjcksn** [comment]
Mark has to approve his change request and the changes from master have to
be rebased before I can merge.

**Comment by chanwit** [comment]
@palemtnrider could you please review this PR again?

**Pull request merged**